File: 백준/Silver/11723. 집합/집합.py
import sys

n = int(sys.stdin.readline())
bit = 0

for _ in range(n):
    func = sys.stdin.readline().split()
    
    if func[0] == "all":
        bit = (1 << 20) -1
    elif func[0] == "empty":
        bit = 0
    
    else:
        num = int(func[1]) - 1

        if func[0] == "add":
            bit |= (1 << num)

        elif func[0] == "remove":
            bit &= ~(1 << num)


        elif func[0] == "check":
            if bit & (1 << num) == 0:
                print(0)
            else:
                print(1)

        elif func[0] == "toggle":
            bit ^= (1 << num)    

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Main {
	
	static String command;
	static int num;
	static int S = 0;
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		StringBuilder sb = new StringBuilder();
		
		int M = Integer.parseInt(br.readLine());
		
		for(int i=0;i<M;i++) {
			st = new StringTokenizer(br.readLine());
			command = st.nextToken();
			
			if(command.equals("all")) S = (1 << 21) - 1; // 최상위 비트만 1인 21자리에서 0b1 빼면 최상위 비트 제외 아래 20자리 모두 1임
			else if(command.equals("empty")) S = 0;
			else {
				num = Integer.parseInt(st.nextToken());
				
				if(command.equals("add")) S |= (1<<num); // OR 연산으로 해당 자리는 무조건 1로 
				else if(command.equals("remove")) S &= ~(1<<num); // 반전 시켜서 해당 자리만 0으로
				else if(command.equals("check")) sb.append(((S &(1<<num)) == 0?0:1)+"\n"); // AND 연산으로 1,1이면 존재 
				else if(command.equals("toggle")) S ^= (1<<num); // XOR로 두 비트가 서로 다르면 1
			}
		}
		System.out.println(sb);
		
		br.close();
		
	}
	
	
	
}
